### ISSUE TITLE
the explorer fetches `swagger.json` multiple times cause the loading very slow

### ISSUE THREAD

**Issue by clarkorz** [strongloop/loopback-component-explorer#129]
After upgrading explorer from 1.8.0 to 2.1.1(by following [upgrading-from-v1x](https://github.com/strongloop/loopback-component-explorer#upgrading-from-v1x)), the explorer takes a very long time to display api interfaces.

It seems that the explorer keeps sending requests to fetch `swagger.json` file, until the file is resolved. The bigger `swagger.json` file is, the more requests it sends, and the more time it takes.

<img width="1280" alt="2015-10-08 6 28 39" src="https://cloud.githubusercontent.com/assets/3261477/10384114/e7dea098-6e69-11e5-90e3-fb8f06676fc3.png">

**Comment by saystone** [comment]
same here.

**Comment by bajtos** [comment]
I was able to reproduce the problem using [loopback-example-app](https://github.com/strongloop/loopback-example-app).

**Comment by bajtos** [comment]
Here is what I found: whenever the swagger.json references a local type (definition) that is actually not included in the swagger file (e.g. GeoPoint in the case of loopback-example-app), swagger-ui attempts to load the swagger file again. As a result, each reference to GeoPoint triggers another swagger.json request.

Does your application happen to use GeoPoint, ObjectID or any other non-defined type? You can tell that by inspecting console logs, explorer prints an error message for each missing definition.

Related: #126 and #116.

**Comment by clarkorz** [comment]
@bajtos I guess you are right. There is a warning as described in #126  . The swagger file will be loaded only once after changing the datasource to memory.

**Comment by bajtos** [comment]
In that case I am closing this as a duplicate of #126. Feel free to reopen if you disagree.

**Comment by mike-aungsan** [comment]
Same issue
